internal/character/seele/talent.go

```diff
@@ -0,0 +1,70 @@
+package seele
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Resurgence key.Modifier = "seele-resurgence"
+)
+
+// Enters the buffed state upon defeating an enemy with Basic ATK, Skill, or Ultimate,
+// and receives an extra turn. While in the buffed state,
+// the DMG of Seele's attacks increases by 80% for 1 turn(s).
+// Enemies defeated in the extra turn provided by "Resurgence"
+// will not trigger another "Resurgence."
+
+func init() {
+	modifier.Register(Resurgence, modifier.Config{
+		Stacking:   modifier.ReplaceBySource,
+		StatusType: model.StatusType_STATUS_BUFF,
+	})
+}
+
+func (c *char) initTalent() {
+	// if seele kills in her turn, set extra turn trigger and add buffedState mod.
+	c.engine.Events().TargetDeath.Subscribe(func(e event.TargetDeath) {
+		if e.Killer == c.id && c.id == c.actionOwner {
+			c.shouldInsert = true
+			c.enterBuffedState()
+		}
+	})
+
+	// actionStart subs to enable adding buffedState mod effectively after-kill
+	c.engine.Events().ActionStart.Subscribe(func(e event.ActionStart) {
+		if !e.IsInsert {
+			c.actionOwner = e.Owner
+		}
+	})
```

Comment:
You need to keep track of if this action is an insert (or at least have some way to check in the `TargetDeath` subscription). Consider the following scenario:
1. Seele ults (ults are not inserts so `actionOwner = seele`)
2. The ult kills, `TargetDeath` causes seele to enter buff state and `shouldInsert` go true
3. At the end of the ult (`ActionEnd`), then `InsertAction` happens
4. (assuming queue only contains the inserted action), `ActionStart` happens for the insert. Because action is an insert, actionOwner is unchanged (`actionOwner = seele` from ult)
5. The insert kills, `TargetDeath` causes seele to enter buff state and `shouldInsert` go true
6. At the end of the ult (`ActionEnd`), then `InsertAction` happens
7. above repeats until Seele doesnt get a kill